vortex-ipc/src/io/object_store.rs

```diff
@@ -0,0 +1,103 @@
+use std::future::Future;
+use std::io::Cursor;
+use std::ops::Range;
+
+use bytes::{Bytes, BytesMut};
+use object_store::path::Path;
+use object_store::{ObjectStore, PutPayload};
+use vortex_buffer::io_buf::IoBuf;
+use vortex_buffer::Buffer;
+use vortex_error::VortexResult;
+
+use crate::io::{VortexRead, VortexReadAt, VortexWrite};
+
+pub trait ObjectStoreExt {
```

Comment:
we just need this simple extension to adapt object_store to what we need in the reader